docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+         *                                        .addConverter(InstantsAsStringsConverter.create())
+         *                                        .build();
+         *
+         *     try (DynamoDbDocumentClient client = DynamoDb.documentClientBuilder()
+         *                                                  .documentClientConfiguration(clientConfiguration)
+         *                                                  .build()) {
+         *
+         *         Table table = client.getTable("my-table");
+         *         UUID id = UUID.randomUUID();
+         *         table.putItem(Item.builder()
+         *                           .putAttribute("partition-key", id)
+         *                           .putAttribute("creation-time", Instant.now())
+         *                           .build());
+         *
+         *         Item item = table.getItem(Item.builder()
+         *                                       .putAttribute("partition-key", id)
+         *                                       .build());
+         *
+         *         // Items are usually stored as a number, but it was stored as an ISO-8601 string now because of the
+         *         // InstantsAsStringsConverter.
+         *         assert item.attribute("creation-time").isString();
+         *         assert item.attribute("creation-time").as(Instant.class).isBetween(Instant.now().minus(1, MINUTE),
+         *                                                                            Instant.now());
+         *     }
+         * </code>
+         */
+        DocumentClientConfiguration.Builder converters(List<ItemAttributeValueConverter<?>> converters);
+        DocumentClientConfiguration.Builder addConverter(ItemAttributeValueConverter<?> converter);
+        DocumentClientConfiguration.Builder clearConverters();
+
+        /**
+         * Create the configuration object client with all of the configured values.
+         */
+        DocumentClientConfiguration build();
+    }
+}
+
+/**
+ * A converter between Java types and DynamoDB types. These can be attached to {@link DynamoDbDocumentClient}s and
+ * {@link Item}s, so that types are automatically converted when writing to and reading from DynamoDB.
+ *
+ * @see DocumentClientConfiguration.Builder#converters(List)
+ * @see Item.Builder#converter(ItemAttributeValueConverter)
+ *
+ * @param T The Java type that is generated by this converter.
+ */
+@ThreadSafe
+public interface ItemAttributeValueConverter<T> {
+    /**
+     * The default condition in which this converter is invoked.
+     *
+     * Even if this condition is not satisfied, it can still be invoked directly via
+     * {@link ItemAttributeValue#convert(ItemAttributeValueConverter)}.
+     */
+    ConversionCondition defaultConversionCondition();
+
+    /**
+     * Convert the provided Java type into an {@link ItemAttributeValue}.
+     */
+    ItemAttributeValue toAttributeValue(T input, ConversionContext context);
```

Comment:
minor: would `? super T` be appropriate here?

Reply:
I don't think so. A converter of `Instant` probably can't convert `Object`s.